### ISSUE TITLE
ResizeSensor expired if appended before an element being appended

### ISSUE THREAD

**Issue by wenchonglai** [marcj/css-element-queries#115]
It appears that the ResizeSensor instances only work as expected after the element it is assigned to has already been appended to the webpage and never been removed. Is this an unresolved issue or there are ways to get away with this?

To reproduce the problem (in Chrome v.51.0.2704.84 (64-bit)), please see the following codes:

1) If a resizeSensor is constructed after an element is appended, the resizeSensor will work.
However, if the element gets removed and appended again, the resizeSensor will NOT work for the following commands within the same script.
However, commands executed asynchronously after the second-time appendChild will work.

```
/* First, run the codes below in a script  */
var div = document.createElement('div');
// This time, do not construct a resizeSensor before appending div to document.body
document.body.appendChild(div);
div.rs2 =  new ResizeSensor( div, function(){ console.log("rs2"); } ); // rs2
div.style.width = '102px'; // rs2
document.body.removeChild(div);
document.body.appendChild(div); // no response
div.style.width = '103px'; // no response
setTimeout( function(){ div.style.width = '104px'; }, 1000); // (after 1s) rs2

/* Then, type the following code in console*/
div.style.width = '105px'; // rs2
```

2) If a resizeSensor is constructed before an element is appended, not only that resizeSensor will not work, the resizeSensors constructed after the element is appended will not work regardless of whether a command is executed synchronously or asynchronously.

```
var div = document.createElement('div');
div.rs = new ResizeSensor( div, function(){ console.log("rs1"); } );
document.body.appendChild(div); // no response
div.style.width = '101px'; // no response
div.rs2 =  new ResizeSensor( div, function(){ console.log("rs2"); } ); // no response
div.style.width = '102px'; // no response
setTimeout( function(){ div.style.width = '104px'; }, 1000); // (after 1s) no response

/* Then, type the following code in console*/
div.style.width = '105px'; // no response
```

**Comment by ShawnSauce** [comment]
An element that has not been appended has zero dimensions and a scrollbars position do not go past the dimensions of the element. Therefore, when the element is appended, the scrollbars positions are zero and are no longer triggering the scroll event.

To fix this, you need to reset the scroll positions after the element has been appended.

It's possible to make a function available in the `element.resizeSensor` variable that will reset the element.

Kind of like this:
```javascript
// Line 152
element.resizeSensor.reset = function () {
    reset();
};

// Then call it like this after appending:
div.resizeSensor.reset();
```

**Comment by ben-page** [comment]
Will you elaborate? How does one "reset the scroll positions"? What does the inner `reset()` call actually do?

**Comment by marcj** [comment]
Fixed in upcoming 1.0.0 version that will be released soon. We also added the `reset` methods.
